Content.Server/Anomaly/Effects/FleshAnomalySystem.cs

```diff
@@ -0,0 +1,98 @@
+        SubscribeLocalEvent<FleshAnomalyComponent, AnomalyStabilityChangedEvent>(OnSeverityChanged);
+    }
+
+    private void OnPulse(EntityUid uid, FleshAnomalyComponent component, ref AnomalyPulseEvent args)
+    {
+        var range = component.SpawnRange * args.Stability;
+        var amount = (int) (component.MaxSpawnAmount * args.Severity + 0.5f);
+
+        var xform = Transform(uid);
+        SpawnMonstersOnOpenTiles(component, xform, amount, range);
+    }
+
+    private void OnSupercritical(EntityUid uid, FleshAnomalyComponent component, ref AnomalySupercriticalEvent args)
+    {
+        var xform = Transform(uid);
+        SpawnMonstersOnOpenTiles(component, xform, component.MaxSpawnAmount, component.SpawnRange);
+        Spawn(component.SupercriticalSpawn, xform.Coordinates);
+    }
+
+    private void OnSeverityChanged(EntityUid uid, FleshAnomalyComponent component, ref AnomalyStabilityChangedEvent args)
+    {
+        var xform = Transform(uid);
+        if (!_map.TryGetGrid(xform.GridUid, out var grid))
+            return;
+
+        var radius = component.SpawnRange * args.Stability;
+        var fleshTile = (ContentTileDefinition) _tiledef[component.FleshTileId];
+        var localpos = xform.Coordinates.Position;
+        var tilerefs = grid.GetLocalTilesIntersecting(
+            new Box2(localpos + (-radius, -radius), localpos + (radius, radius)));
+        foreach (var tileref in tilerefs)
+        {
+            if (!_random.Prob(0.33f))
+                continue;
+            _tile.ReplaceTile(tileref, fleshTile);
+        }
+    }
+
+    private void SpawnMonstersOnOpenTiles(FleshAnomalyComponent component, TransformComponent xform, int amount, float radius)
+    {
+        if (!_map.TryGetGrid(xform.GridUid, out var grid))
+            return;
+
+        var localpos = xform.Coordinates.Position;
+        var tilerefs = grid.GetLocalTilesIntersecting(
+            new Box2(localpos + (-radius, -radius), localpos + (radius, radius))).ToArray();
+        _random.Shuffle(tilerefs);
+        var physQuery = GetEntityQuery<PhysicsComponent>();
+        var amountCounter = 0;
+        foreach (var tileref in tilerefs)
+        {
+            var valid = true;
+            foreach (var ent in grid.GetAnchoredEntities(tileref.GridIndices))
+            {
+                if (!physQuery.TryGetComponent(ent, out var body))
+                    continue;
+                if (body.BodyType != BodyType.Static ||
+                    !body.Hard ||
+                    (body.CollisionLayer & (int) CollisionGroup.Impassable) == 0)
+                    continue;
+                valid = false;
+                break;
+            }
+
+            if (!valid)
+                continue;
+            amount++;
+            Spawn(_random.Pick(component.Spawns), tileref.GridIndices.ToEntityCoordinates(xform.GridUid.Value, _map));
+            if (amountCounter >= amount)
+                return;
+        }
```

Comment:
disclaimer: this may or may not work and indentation may or may not be off to keep you on your toes (btw neither mine or yours account for equalizing spawns across tiles cuz if you have 9 tiles and 10 spawns you would spawn 9 monsters, but it think its fine for now)
```suggestion
        var tilerefs = grid.GetLocalTilesIntersecting(
            new Box2(localpos + (-radius, -radius), localpos + (radius, radius)));
        _random.Shuffle(tilerefs);
        var physQuery = GetEntityQuery<PhysicsComponent>();
        foreach (var tileref in tilerefs)
        {
            bool spawned = false;
            foreach (var ent in grid.GetAnchoredEntities(tileref.GridIndices))
            {
                if (spawned || amount <= 0)
                    break;

                if (!physQuery.TryGetComponent(ent, out var body))
                    continue;

                if (body.BodyType != BodyType.Static &&
                    !body.Hard &&
                    (body.CollisionLayer & (int) CollisionGroup.Impassable) == 0) // idk if this is correct, i despise bit flags and basic boolean math
                {
                    var selectedSpawn = _random.Pick(component.Spawns); // thinking about it, why not shuffle component.Spawns on severity change or something instead?
                    Spawn(selectedSpawn, tileref.GridIndices.ToEntityCoordinates(xform.GridUid.Value, _map));
                    spawned = true;
                    amount--
                }
            }
        }
```